R/shinyui.R

```diff
@@ -80,6 +60,40 @@ renderPage <- function(ui, showcase=0, testMode=FALSE) {
   enc2utf8(paste(collapse = "\n", html))
 }
 
+shinyDependencies <- tagFunction(function() {
+  cssFile <- shinyCssFile()
+
+  version <- utils::packageVersion("shiny")
+  list(
+    htmlDependency(
+      name = "shiny-css",
+      version = version,
+      src = cssFile$src,
+      stylesheet = cssFile$stylesheet
+    ),
+    htmlDependency(
+      name = "shiny-javascript",
+      version = version,
+      src = c(href = "shared"),
+      script = if (getOption("shiny.minified", TRUE)) "shiny.min.js" else "shiny.js"
+    )
+  )
+})
+
+shinyCssFile <- function(theme = getShinyOption("bootstrapTheme")) {
+  if (!is_bs_theme(theme)) {
+    return(list(src = c(href = "shared"), stylesheet = "shiny.css"))
+  }
+  scss_home <- system.file("www", "shared", "shiny_scss", package = "shiny")
+  scss_files <- file.path(scss_home, c("bootstrap.scss", "shiny.scss"))
+  outFile <- bootstrapSass(lapply(scss_files, sass::sass_file), theme = theme, "shiny")
+  list(src = c(file = dirname(outFile)), stylesheet = basename(outFile))
+}
+
+jqueryDependency <- function() {
+  jquerylib::jquery_core(getOption("shiny.jquery.version", 3))
```

Comment:
I think we should do the jquerylib stuff in a separate PR. This change points to a copy of jQuery on a CDN, which will cause breakage for people who are on private networks.

```
> jquerylib::jquery_core(getOption("shiny.jquery.version", 3))
List of 10
 $ name      : chr "jquery"
 $ version   : chr "3.5.1"
 $ src       :List of 2
  ..$ href: chr "https://code.jquery.com"
  ..$ file: chr "lib"
 $ meta      : NULL
 $ script    : chr "jquery-3.5.1.min.js"
 $ stylesheet: NULL
 $ head      : NULL
 $ attachment: NULL
 $ package   : chr "jquerylib"
 $ all_files : logi TRUE
 - attr(*, "class")= chr "html_dependency"
```

Reply:
Oh interesting, I had thought that `file` took precedence over `href` -- if that's not true, then I'll remove the `href` in **jquerylib**

Reply:
(We want to make sure we're doing the right thing in **jquerylib** because **bootstraplib** imports it)

Reply:
I still think we should revert this stuff and do the jquerylib stuff in another PR, so that we can focus on the details of that in one place (for example, possibly removing jquery files from shiny)